types/node_credentials.go

```diff
@@ -317,6 +338,26 @@ func NewNodeCredentials(
 	return n, nil
 }
 
+// SetPreviousEncryptionKey will set this NodeCredential's PreviousEncryptionKey field
+// using the passed NodeCredentials
+func (n *NodeCredentials) SetPreviousEncryptionKey(oldNodeCredentials *NodeCredentials) error {
+	const op = "nodeenrollment.types.(NodeCredentials).SetPreviousEncryptionKey"
+	if oldNodeCredentials == nil {
+		return fmt.Errorf("(%s) empty prior credentials passed in", op)
+	}
+
+	previousEncryptionKey := &EncryptionKey{
+		Id:              oldNodeCredentials.Id,
```

Comment:
I believe that ID will always be "current" and that generally the id we're using via passing into the functions is derived from the public certificate key. It's hard to track...this is a reason I'm actually thinking of having id be part of the normal interface too. Then for anything implementing the interface it's easy to see what the ID will come from.

I may go ahead and work on that change concurrently so it will be easier to validate correctness for previous key.